magda-web-client/src/Components/Common/DataPreviewMap.tsx

```diff
@@ -0,0 +1,325 @@
+import React, { Component, useEffect, useState } from "react";
```

Comment:
Sorry about the full-page diff here - the key changes are that introduced `singleFile` to the array of data sources - if this is set to true, the file will be checked for length before it's allowed to be previewed. If it's false, it's assumed that the dataset is a service that can't be reasonably checked for data volume, and it's just allowed to be previewed.